### ISSUE TITLE
macOS: Permission denied error on reboot

### ISSUE THREAD

**Issue by micahflee** [firstlookmedia/flock-agent#101]
This is happening in Flock Agent 0.1.3, even though I thought this issue was fixed. It only happens on reboot, and then Flock Agent appears to work fine after.

<img width="428" alt="Screen Shot 2020-01-24 at 9 57 16 AM" src="https://user-images.githubusercontent.com/156128/73091805-0d71dc80-3e90-11ea-9786-8e142b0467ce.png">

**Comment by micahflee** [comment]
Here's the log on reboot:

```
2020-01-24 11:38:33 Bootstrap.__init__
2020-01-24 11:38:33 Bootstrap.go: Bootstrapping Flock Agent
2020-01-24 11:38:33 Bootstrap.go: Making sure Flock Agent starts automatically
2020-01-24 11:38:33 Bootstrap.go: Making sure osquery is installed
2020-01-24 11:38:33 Bootstrap.go: Making sure the Flock Agent daemon is running
2020-01-24 11:38:33 DaemonClient._request: get /ping
2020-01-24 11:38:33 Bootstrap.go: Permission denied ...
2020-01-24 11:38:34 DaemonClient._request: get /ping
2020-01-24 11:38:34 Bootstrap.go: Permission denied ...
2020-01-24 11:38:35 DaemonClient._request: get /ping
2020-01-24 11:38:35 Bootstrap.go: Permission denied ...
2020-01-24 11:38:36 DaemonClient._request: get /ping
2020-01-24 11:38:36 Bootstrap.go: Permission denied ...
2020-01-24 11:38:37 DaemonClient._request: get /ping
2020-01-24 11:38:37 Bootstrap.go: Permission denied ...
2020-01-24 11:38:38 GuiCommon.daemon_permission_denied
```

And meanwhile, the log for the daemon:

```
2020-01-24 11:39:54 Daemon.autoupdate_loop: Checking for updates
2020-01-24 11:39:55 Daemon.autoupdate_loop: installed version: 0.1.3, latest version: 0.1.3
2020-01-24 11:39:55 Daemon.http_server: Started http server
```

Notice that the agent finally gave up talking to the daemon at 11:38:38, but the daemon's http service didn't start until 11:39:55.

The problem here is that the agent gives up before the daemon has started.

The naive solution is to just give the agent more time to try to connect (10 seconds instead of 5, perhaps). I should do this anyway, but I think the root cause of the issue is that the agent is getting a permission denied error when trying to communicate with a unix socket that nothing is listening on.

Really the exception it should be handling is `DaemonNotRunningException`, not `PermissionDeniedException`. In which case instead of popping up the permission denied error, it will pop up a more reasonable daemon not running error. I need to make it so when the daemon stops, it deletes its unix socket file.

**Comment by micahflee** [comment]
I've discovered another issue. On my Mac, internet gets blocked until my VPN successfully connects, which makes the autoupdate check throw exceptions (it can't load github.com), and loop forever trying over and over again until it can successfully connect. 

But because we're using asyncio and not threads, and because the autoupdate loop does `await` anything, the http server is actually being blocked by the autoupdate function. So I need to improve the autoupdate function and make it `await` stuff, to give other parts of the daemon a chance to start.
